src/classes/RD2_ScheduleService_TEST.cls

```diff
@@ -1017,6 +1017,56 @@ private with sharing class RD2_ScheduleService_TEST {
         System.assertEquals(0, schedules.size(), 'There should be zero schedules.');
     }
 
+    /****
+    * @description Validates start date on RD with unmodified past start date
+    */
+    @IsTest
+    private static void shouldApplyRDChangesIfPastStartDateUnchanged() {
```

Comment:
The test should be verifying the RD submitted to the update trigger is marked for update and Effective Date is unchanged? If so, the ApexDoc description and test can be written as:
```
    /****
    * @description Validates the RD past Effective Date is unchanged when Amount is changed 
    * and schedule is marked for update.
    */
    @IsTest
    private static void shouldMarkRDForScheduleUpdateWhenStartDateIsInPast() {
        Date startDate = Date.newInstance(2019, 11, 1);
        RD2_EnablementService_TEST.setRecurringDonations2Enabled();

        List<npe03__Recurring_Donation__c> oldRds = new List<npe03__Recurring_Donation__c>{
            getRecurringDonationMonthlyBuilder()
                .withStartDate(startDate)
                .withMockId()
                .build()
        };

        Decimal newAmount = 250;
        List<npe03__Recurring_Donation__c> updatedRds = new List<npe03__Recurring_Donation__c>{
            oldRds[0].clone(true)
        };
        updatedRds[0].npe03__Amount__c = newAmount;
        
        RD2_ScheduleService service = new RD2_ScheduleService();
        Map<Id, npe03__Recurring_Donation__c> rdsNeedUpdate = service.getRecurringDonationsNeedingScheduleUpdates(
            updatedRds, oldRds, TDTM_Runnable.Action.BeforeUpdate
        );

        System.assertEquals(1, rdsNeedUpdate.size(), 'Recurring Donation should be marked for update');
        System.assertEquals(newAmount, rdsNeedUpdate.get(rd.Id).npe03__Amount__c, 'Updated amount should match');
        System.assertEquals(startDate, rdsNeedUpdate.get(rd.Id).StartDate__c, 'RD Effective Date should be unchanged');
    }
```